File: wrappers/test/js_common/test/vc/oid4vp/mockNonceHandler.ts
import { NonceHandler } from "agent-sdk";

export class MockNonceHandler implements NonceHandler {
  /** The nonces of each `invalidate` call, in call order. */
  readonly invalidated: Array<Array<string>> = [];

  constructor(private readonly nonce: string) {
    this.generate = this.generate.bind(this);
    this.validate = this.validate.bind(this);
    this.invalidate = this.invalidate.bind(this);
  }

  async generate(): Promise<string> {
    return this.nonce;
  }

  async validate(nonce: string): Promise<boolean> {
    return true;
  }

  async invalidate(nonces: Array<string>): Promise<void> {
    this.invalidated.push(nonces);
  }
}

File: wrappers/uniffi/swift/Tests/AsdkTests/MockNonceHandler.swift
@testable import Asdk

final class MockNonceHandler : NonceHandler {
    private let nonce: String
    init(nonce: String) {
        self.nonce = nonce
    }
    func generate() async throws -> String {
        return self.nonce
    }

    func validate(nonce: String) -> Bool {
        return true
    }
}